test/functional/lua/buffer_updates_spec.lua

```diff
@@ -420,6 +420,35 @@ describe('lua buffer event callbacks: on_lines', function()
     feed('i<Tab>')
     eq({ '\ta' }, exec_lua('return _G.res[#_G.res]'))
   end)
+
+  it('quickfix buffer send change', function()
```

Comment:
this test file is for the lua interface to bufupdates, whereas the fix is a low-level change, so probably fits better in `api/buffer_updates_spec`. perhaps after the diff-mode test here: https://github.com/neovim/neovim/blob/76d213efbee8e6857b3aa397827fa8756a327430/test/functional/api/buffer_updates_spec.lua#L584

#31085 mentions `on_bytes`, but the fix in this PR is simply to enable buffer-updates, so there is no need to test all the "lua callback" functionality. Instead we just need to verify that buffer updates are working in quickfix bufs.

Reply:
extmark_splice called `buf_updates_send_splice` which trigger `on_bytes`